auto_route/lib/src/router/widgets/auto_page_view.dart

```diff
@@ -81,19 +84,21 @@ class AutoPageViewState extends State<AutoPageView> {
   }
 
   Future<void> _warpToCurrentIndex() async {
-    if (!mounted) return Future<void>.value();
+    final Duration duration = widget.duration;
+    final bool animatePageTransition = widget.animatePageTransition;
 
-    const Duration duration = Duration(milliseconds: 300);
+    if (!mounted) return Future<void>.value();
 
-    if (duration == Duration.zero) {
+    if (!animatePageTransition || duration == Duration.zero) {
       _controller.jumpToPage(_router.activeIndex);
       return Future<void>.value();
     }
+
     final int previousIndex = _router.previousIndex ?? 0;
     if ((_router.activeIndex - previousIndex).abs() == 1) {
       _warpUnderwayCount += 1;
       await _controller.animateToPage(_router.activeIndex,
-          duration: duration, curve: Curves.ease);
+          duration: widget.duration, curve: Curves.ease);
       _warpUnderwayCount -= 1;
       return Future<void>.value();
```

Comment:
@Milad-Akarie is there a particular reason for returns like that?
I used the same return types but maybe use only `return;` is better.
If you agree, i can do another PR where i change all returns like that